pinot-plugins/pinot-input-format/pinot-parquet/src/main/java/org/apache/pinot/plugin/inputformat/parquet/ParquetNativeRecordExtractor.java

```diff
@@ -143,14 +142,26 @@ private Object extractValue(Group from, int fieldIndex) {
   }
 
   private Object extractValue(Group from, int fieldIndex, Type fieldType, int index) {
-    OriginalType originalType = fieldType.getOriginalType();
+    LogicalTypeAnnotation logicalTypeAnnotation = fieldType.getLogicalTypeAnnotation();
     if (fieldType.isPrimitive()) {
       PrimitiveType.PrimitiveTypeName primitiveTypeName = fieldType.asPrimitiveType().getPrimitiveTypeName();
       switch (primitiveTypeName) {
         case INT32:
-          return from.getInteger(fieldIndex, index);
+          int intValue = from.getInteger(fieldIndex, index);
+          if (logicalTypeAnnotation instanceof LogicalTypeAnnotation.DecimalLogicalTypeAnnotation) {
```

Comment:
we can do the following:
```
if (logicalTypeAnnotation.getType() == LogicalTypeToken.DECIMAL)
```

Reply:
I don't find this `logicalTypeAnnotation.getType()` api

Reply:
![Screenshot 2023-10-20 at 12 59 53 PM](https://github.com/apache/pinot/assets/27253407/d30cc0e2-8ba7-4f25-b21b-bfcde03e7034)
 I saw `abstract LogicalTypeToken getType()` but looks that it doesn't have `public` access so it's not usable outside.